cmd/security-secrets-setup/main.go

```diff
@@ -69,24 +75,52 @@ func main() {
 		return
 	}
 
-	if configFile == "" {
-		// run with other options for pki-init
-		statusCode, err := dispatcherInstance.run()
-		if err != nil {
-			setup.LoggingClient.Error(err.Error())
-		}
+	if checkIfMultipleSubcommands() {
+		setup.LoggingClient.Error("cannot use multiple subcommands, use one at a time")
+		exitInstance.exit(1)
+		return
+	}
+
+	subcmdName := flag.Args()[0]
 
-		exitInstance.exit(statusCode)
+	subcmd, found := subcommands[subcmdName]
+	if !found {
+		setup.LoggingClient.Error(fmt.Sprintf("unsupported subcommand %s", subcmdName))
+		exitInstance.exit(1)
 		return
 	}
 
-	if err := option.GenTLSAssets(configFile); err != nil {
-		setup.LoggingClient.Error(err.Error())
+	if err := subcmd.Parse(flag.Args()[1:]); err != nil {
+		setup.LoggingClient.Error(fmt.Sprintf("error parsing subcommand %s: %v", subcmdName, err))
 		exitInstance.exit(2)
 		return
 	}
 
+	var exitStatusCode int
+	var err error
+
+	switch subcmdName {
+	case "legacy":
+		if err = option.GenTLSAssets(configFile); err != nil {
+			setup.LoggingClient.Error(err.Error())
+			exitInstance.exit(2)
+			return
+		}
+	case "generate", "cache", "import":
```

Comment:
since we don't support `cache` and `import` yet can we drop these in this PR?

Reply:
ok